synapse/storage/databases/main/search.py

```diff
@@ -588,26 +596,27 @@ async def search_rooms(
                 raise SynapseError(400, "Invalid pagination token")
 
             clauses.append(
-                "(origin_server_ts < ?"
-                " OR (origin_server_ts = ? AND stream_ordering < ?))"
+                """(origin_server_ts < ?
+                OR (origin_server_ts = ? AND stream_ordering < ?))
```

Comment:
Aside: I wonder if this can be expressed as `( (origin_server_ts, stream_ordering) < (?, ?) )` using a lexicographic ordering?